dace/transformation/auto/auto_optimize.py

```diff
@@ -387,6 +397,11 @@ def set_fast_implementations(sdfg: SDFG, device: dtypes.DeviceType, blocklist: L
     # general nodes
     for node, _ in sdfg.all_nodes_recursive():
         if isinstance(node, nodes.LibraryNode):
+            # NOTE: LibraryNodes with sequential schedule on GPU must be expanded to CUDA kernel-compatible code.
+            # NOTE: Pure implementations are a safe choice for now but this should be revisited in the future.
+            if device == dtypes.DeviceType.GPU and node.schedule == dtypes.ScheduleType.Sequential:
+                node.implementation = "pure"
```

Comment:
for some reason it doesn't seem like the "right" fix. Maybe default `implementation` could depend on the device context it's executing in.